src/VisualStudio/Core/Def/Experimentation/VisualStudioExperimentationService.cs

```diff
@@ -10,7 +11,7 @@
 namespace Microsoft.VisualStudio.LanguageServices.Experimentation
 {
     [ExportWorkspaceService(typeof(IExperimentationService), ServiceLayer.Host), Shared]
-    internal class VisualStudioExperimentationService : IExperimentationService
+    internal class VisualStudioExperimentationService : ForegroundThreadAffinitizedObject, IExperimentationService
```

Comment:
I think there is a way to mark that this service needs to be created from foreground thread. I believe one of base contructor accepts flag for it?